### PR TITLE
docs: document PackManifestOptions to make PackManifest reproducible

### LINKED ISSUE #748: docs: document `PackManifestOptions` to make `PackManifest` reproducible
[PackManifest](https://pkg.go.dev/oras.land/oras-go/v2#PackManifest) packs an artifact by generating a manifest with annotation `org.opencontainers.image.created`.

Each time calling `PackManifest`, a new time stamp is generated for `org.opencontainers.image.created` even with the same content. Therefore, building an artifact is not [reproducible](https://en.wikipedia.org/wiki/Reproducible_builds).

The `PackManifest` method can be _reproducible_ when `ocispec.AnnotationCreated` is set. However, this fact is not called out in the docs of [PackManifestOptions](https://pkg.go.dev/oras.land/oras-go/v2#PackManifestOptions) as well as [PackManifest](https://pkg.go.dev/oras.land/oras-go/v2#PackManifest).

This request is to enhance the documentation for reproducibility of `PackManifest`.

### PR DESCRIPTION
**Pull request by wangxiaoxuan273**
Resolves #748

### CODE REVIEW

File: pack.go [hunk 1]
```diff
@@ -126,6 +128,11 @@ var mediaTypeRegexp = regexp.MustCompile(`^[A-Za-z0-9][A-Za-z0-9!#$&-^_.+]{0,126
 //
 // artifactType and opts.ConfigDescriptor.MediaType MUST comply with RFC 6838.
 //
+// Each time when PackManifest is called, it generates a new time stamp in the
+// manifest annotations with the key "org.opencontainers.image.created". To make
```
Comment (Wwwsylvia) [inline pack.go:132]: We also need to mention that if "org.opencontainers.image.created" is specified, it must conform to [RFC 3339](https://www.rfc-editor.org/rfc/rfc3339.html).

File: pack.go [hunk 2]
```diff
@@ -93,7 +93,9 @@ type PackManifestOptions struct {
 	// Layers is the layers of the manifest.
 	Layers []ocispec.Descriptor
 
-	// ManifestAnnotations is the annotation map of the manifest.
+	// ManifestAnnotations is the annotation map of the manifest. Set
+	// "org.opencontainers.image.created" with a value to make PackManifest
+	// reproducible.
```
Comment (Wwwsylvia) [inline pack.go:98]: The timestamp must be in [RFC 3339](https://www.rfc-editor.org/rfc/rfc3339.html) format.

File: pack.go
Comment (Wwwsylvia) [inline pack.go:1]: We may also want to update the doc of `ErrInvalidDateTimeFormat` and replace `AnnotationCreated ` with the literal string "org.opencontainers.image.created".

**Review by Wwwsylvia — CHANGES_REQUESTED** [review]
We also need to rephrase the related comment a bit for `ExamplePackManifest_imageV10` and `ExamplePackManifest_imageV11` in `example_pack_test.go`.

**Comment by Wwwsylvia** [comment]
BTW, the PR title should start with `docs` (end with `s`) as per the [conventional commit guidance](https://www.conventionalcommits.org/en/v1.0.0/#summary). 😀

File: pack.go [hunk 3]
```diff
@@ -93,7 +93,9 @@ type PackManifestOptions struct {
 	// Layers is the layers of the manifest.
 	Layers []ocispec.Descriptor
 
-	// ManifestAnnotations is the annotation map of the manifest.
+	// ManifestAnnotations is the annotation map of the manifest. Set
+	// "org.opencontainers.image.created" with a value to make PackManifest
```
Comment (shizhMSFT) [inline pack.go:97]: Developers is not encouraged to use string literals but constants. Please refer the developers to use constants from ocispec.

File: pack.go [hunk 4]
```diff
@@ -126,6 +128,11 @@ var mediaTypeRegexp = regexp.MustCompile(`^[A-Za-z0-9][A-Za-z0-9!#$&-^_.+]{0,126
 //
 // artifactType and opts.ConfigDescriptor.MediaType MUST comply with RFC 6838.
 //
+// Each time when PackManifest is called, it generates a new time stamp in the
+// manifest annotations with the key "org.opencontainers.image.created". To make
+// PackManifest reproducible, set "org.opencontainers.image.created" with a value
```
Comment (shizhMSFT) [inline pack.go:133]: Same here for consts.

Comment (shizhMSFT) [reply pack.go:133]: This comment is a duplicate of https://github.com/oras-project/oras-go/pull/749/files#r1575970704

File: pack.go (hunk 3 above)
Comment (shizhMSFT) [reply pack.go:97]: This comment is a duplicate of https://github.com/oras-project/oras-go/pull/749/files#r1575970704

**Review by shizhMSFT — COMMENTED** [review]
You also need to update the examples in `example_pack_test.go`.

File: pack.go (hunk 3 above)
Comment (Wwwsylvia) [reply pack.go:97]: Actually we are talking about different stuffs😀 In [another comment](https://github.com/oras-project/oras-go/pull/749#discussion_r1575970704) my concern was that users may not understand what `AnnotationArtifactCreated` and `AnnotationCreated` are as they are not defined in `oras-go`, and we may want to use string literals in the comment.

https://github.com/oras-project/oras-go/blob/6fce64e956e850a3af7ea16e85959c23e3eddc4b/pack.go#L50-L54

File: pack.go (hunk 1 above)
Comment (wangxiaoxuan273) [reply pack.go:132]: Added a comment.

File: pack.go (hunk 2 above)
Comment (wangxiaoxuan273) [reply pack.go:98]: Added a comment.

File: pack.go
Comment (wangxiaoxuan273) [reply pack.go:1]: Replaced both `AnnotationArtifactCreated` and `AnnotationCreated` with the corresponding literal strings.

File: pack.go (hunk 3 above)
Comment (wangxiaoxuan273) [reply pack.go:97]: > Actually we are talking about different stuffs😀 In [another comment](https://github.com/oras-project/oras-go/pull/749#discussion_r1575970704) my concern was that users may not understand what `AnnotationArtifactCreated` and `AnnotationCreated` are as they are not defined in `oras-go`, and we may want to use string literals in the comment.
> 
> https://github.com/oras-project/oras-go/blob/6fce64e956e850a3af7ea16e85959c23e3eddc4b/pack.go#L50-L54

Has a point. Kept using the string literal in the comment for the error value, and replaced with the variable name `ocispec.AnnotationCreated` for comments of the function and annotation map.

File: pack.go (hunk 4 above)
Comment (wangxiaoxuan273) [reply pack.go:133]: Replaced the const with the variable name.

**Comment by wangxiaoxuan273** [comment]
> We also need to rephrase the related comment a bit for `ExamplePackManifest_imageV10` and `ExamplePackManifest_imageV11` in `example_pack_test.go`.

updated the comments  in `example_pack_test.go`.

File: example_pack_test.go [hunk 5]
```diff
@@ -35,7 +35,8 @@ func ExamplePackManifest_imageV11() {
 	opts := oras.PackManifestOptions{
 		ManifestAnnotations: map[string]string{
 			// this timestamp will be automatically generated if not specified
-			// use a fixed value here in order to test the output
+			// use a fixed value here in order to test the output and make the
```
Comment (Wwwsylvia) [inline example_pack_test.go:38]: nit: How about:
```suggestion
			// use a fixed value here to make the pack result reproducible
```
?

File: pack.go [hunk 6]
```diff
@@ -126,6 +128,11 @@ var mediaTypeRegexp = regexp.MustCompile(`^[A-Za-z0-9][A-Za-z0-9!#$&-^_.+]{0,126
 //
 // artifactType and opts.ConfigDescriptor.MediaType MUST comply with RFC 6838.
 //
+// Each time when PackManifest is called, it generates a new time stamp in the
+// manifest annotations with the key ocispec.AnnotationCreated. To make
```
Comment (Wwwsylvia) [inline pack.go:132]: nit: Do we need to explicitly call out that the timestamp is generated only when `ocispec.AnnotationCreated` annotation is not specified?

File: pack.go [hunk 7]
```diff
@@ -93,7 +93,9 @@ type PackManifestOptions struct {
 	// Layers is the layers of the manifest.
 	Layers []ocispec.Descriptor
 
-	// ManifestAnnotations is the annotation map of the manifest.
+	// ManifestAnnotations is the annotation map of the manifest. In order to
+	// make PackManifest reproducible, set the key ocispec.AnnotationCreated to
```
Comment (shizhMSFT) [inline pack.go:97]:
```suggestion
	// make [PackManifest] reproducible, set the key ocispec.AnnotationCreated to
```

File: pack.go
Comment (shizhMSFT) [reply pack.go:1]: Consider the wording like `ocispec.AnnotationCreated (i.e. "org.opencontainers.image.created")`

File: pack.go (hunk 6 above)
Comment (wangxiaoxuan273) [reply pack.go:132]: Modified the comment

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by Wwwsylvia — APPROVED** [review]
LGTM

**Review by Wwwsylvia — APPROVED** [review]
LGTM

**Pull request merged**